### PR TITLE
Implement fast-path of super expression in static field initializer

### PR DESCRIPTION
**Pull request by ksh8281**
Signed-off-by: Seonghyun Kim <sh8281.kim@samsung.com>

### CODE REVIEW

File: src/interpreter/ByteCode.h [hunk 1]
```diff
@@ -2268,7 +2269,7 @@ class OpenLexicalEnvironment : public ByteCode {
         m_endPostion = SIZE_MAX;
     }
 
-    Kind m_kind : 2;
+    Kind m_kind : 4;
```
Comment (clover2123) [inline src/interpreter/ByteCode.h:2272]: Does `Kind` need 4bits to represent the value?
At the current status, `Kind` has 4 values which could be handled by 2bits.

File: src/interpreter/ByteCodeInterpreter.cpp [hunk 2]
```diff
@@ -2867,6 +2871,8 @@ NEVER_INLINE Value ByteCodeInterpreter::openLexicalEnvironment(ExecutionState*&
     bool shouldUseHeapAllocatedState = true;
     if (inPauserResumeProcess) {
         shouldUseHeapAllocatedState = false;
+    } else if (code->m_kind == OpenLexicalEnvironment::ClassStaticFieldInit) {
+        shouldUseHeapAllocatedState = false;
```
Comment (clover2123) [inline src/interpreter/ByteCodeInterpreter.cpp:2875]: How about merging these two conditional statements into one if statement?

File: src/interpreter/ByteCode.h (hunk 1 above)
Comment (ksh8281) [reply src/interpreter/ByteCode.h:2272]: I will add some values when implemting class field value init

Comment (clover2123) [reply src/interpreter/ByteCode.h:2272]: okay :)

File: src/interpreter/ByteCodeInterpreter.cpp (hunk 2 above)
Comment (ksh8281) [reply src/interpreter/ByteCodeInterpreter.cpp:2875]: That's good idea

Comment (ksh8281) [reply src/interpreter/ByteCodeInterpreter.cpp]: I fixed it

**Review by clover2123 — APPROVED** [review]
LGTM

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**